src/block/block_ext.c

```diff
@@ -414,14 +414,37 @@ __wt_block_off_remove_overlap(
     if (before != NULL && before->off + before->size > off) {
         WT_RET(__block_off_remove(session, block, el, before->off, &ext));
 
+        WT_ASSERT(session, ext->off + ext->size >= off + size);
+
         /* Calculate overlapping extents. */
         a_off = ext->off;
         a_size = off - ext->off;
         b_off = off + size;
         b_size = ext->size - (a_size + size);
+
+        if (a_size > 0 && b_size > 0) {
```

Comment:
If I understand correctly, this change intends to improve the understanding of the underlying system. If that's the case, this is a lot of infrastructure just for debug logs. We need to find a balance between keeping this function simple and readable as well as providing useful logs. Both are important in supportability and debugability. In that sense, I think we could add a single debug log here. Something along the lines of 
```suggestion
                __wt_verbose(session, WT_VERB_BLOCK,
          "%s: %" PRIdMAX "-%" PRIdMAX " remove range %" PRIdMAX "-%" PRIdMAX, el->name,
          (intmax_t)before->off, (intmax_t)before->off + (intmax_t)before->size, (intmax_t)off,
          (intmax_t)off + (intmax_t)size);
```
Does this align with what you would like to observe in the logs?

Reply:
yes, another purpose is to make it easy for beginners to quickly understand the code.